### PR TITLE
[CI:DOCS] Add get_ci_vm script-interface documentation

### PR DESCRIPTION
**Pull request by cevich**
For too long, most of this has had a single source: My Skull.  Attempt to spell it out for others benefit (maybe).

### CODE REVIEW

**Comment by cevich** [comment]
@lsm5 and @edsantiago PTAL if you have time (low-priority).  This is not critical, and I have proof-read it a few times as best as I could.  I don't think it will be a problem to simply merge as-is, un-reviewed.  I doubt anybody will ever actually read it, but will hopefully be somewhat valuable if it is needed.

File: get_ci_vm/README.md [hunk 1]
```diff
@@ -1,10 +1,151 @@
...
+The container expects to be called in a specific way, by a script which
+itself needs to behave according to a set of rules (outlined below).
+This bidirectional requirement scheme is necessary to properly support
+differing runtime details and requirements across multiple repositories.
+
+## Downstream repository `.cirrus.yml` requirements
+
+1. *Must* be valid YAML parsable by a **standard** parser.
+1. *Must* not rely on [a `Starlark` script](https://cirrus-ci.org/guide/programming-tasks/).
+1. *Must* not have any conflicting/overlaping task or alias names.  If a matrix
```
Comment (edsantiago) [inline get_ci_vm/README.md:26]: My eye finds that confusing: stress on *Must*, to me, means affirmation. I'd really prefer stress on both: *Must not*. (Other instances below).

File: get_ci_vm/README.md [hunk 2]
```diff
@@ -1,10 +1,151 @@
...
+   passed as-is into the container as arguments, with the following exceptions:
+   1. When called with the `--setup` option, the script should:
+      1. Confirm `$GET_CI_VM` is non-zero.  This signifies the script is (now)
+         executing from **within** a newly created VM with CWD undefined.
+      1. Execute any commands necessary (on the VM) to prepare it for
+         use by the user.  For example, this could include calling *make* targets
+         and/or environment-setup scripts like `./contrib/cirrus/setup.sh`.
+      1. Exit non-zero if there is an error.
+   1. When called with the `--config` option, the script *must*:
+      1. Confirm `$GET_CI_VM` is `1`.  This signifies the script is (now)
```
Comment (edsantiago) [inline get_ci_vm/README.md:98]: Line 91 says "non-zero". The code (as best I can tell) also just checks for nonzero. If this comment is accurate, could you clarify why `1` is special and different from nonzero?

File: get_ci_vm/README.md [hunk 3]
```diff
@@ -1,10 +1,151 @@
...
+         executing from **within** the get_ci_vm container context.
+      1. Print on *stdout*, a list of `key=value` env. var lines conforming
+         to [the *APIV1 Env. Vars.* section below](README.md#env-vars).
+      1. Exit non-zero if there is an error.
+
+### Env. Vars.
+
+When `hack/get_ci_vm.sh` is executed with the `--config` argument, it
+*must* emit a `key=value` list of env. vars. to *stdout* if and only if
+the `$GET_CI_VM` value is set to `1`.
```
Comment (edsantiago) [inline get_ci_vm/README.md:108]: Again, I don't see any code checking for `1`, just for nonzero.

File: get_ci_vm/README.md [hunk 4]
```diff
@@ -1,10 +1,151 @@
...
+
+* `DESTDIR` - *Must* be set to the "clone" directory (on the VM) for the
+  repository's code.  This directory is analogous to the `$CIRRUS_WORKING_DIR`
+  value in `.cirrus.yml`.  It will created if it doesn't exist, and
```
Comment (edsantiago) [inline get_ci_vm/README.md:112]: (minor) "it will >>BE<< created"

File: get_ci_vm/README.md [hunk 5]
```diff
@@ -1,10 +1,151 @@
...
+  For example, the podman repository's `./contrib/cirrus/setup_environment.sh`
+  populates a `/etc/ci_environment` file with important CI setup-time values.
+* `GCLOUD_PROJECT` - *Must* contain the name of the GCE project space where
+  the VM will be created.
+* `GCLOUD_IMGPROJECT` - *Must* be set to `libpod-218412` (the GCE project space
+  containing all the VM images).
+* `GCLOUD_CFG` - *Must* be set to the name of the GCE configuration to maintain
+  for this repository.  See the GCE settings bind-mount item under
+  the *Operational Details* section above.
+* GCLOUD_ZONE - *Must* be set to either `us-central1-a` or any valid network-near
```
Comment (edsantiago) [inline get_ci_vm/README.md:130]: Missing backticks

File: get_ci_vm/README.md [hunk 6]
```diff
@@ -1,10 +1,151 @@
...
+  zone to the user.   Valid values are listed in [the
+  documentation](https://cloud.google.com/compute/docs/regions-zones/#available)
```
Comment (edsantiago) [inline get_ci_vm/README.md:132]: (minor) "the documentation" is vague; a user will have to hover or click to find out _what_ documentation. Maybe "gcloud documentation" or "google region zone documentation"?

File: get_ci_vm/README.md [hunk 7]
```diff
@@ -1,10 +1,151 @@
...
+* `GCLOUD_DISK` - *Must* be set to `200` (gigabytes).
+
+## APIv2 (AWS EC2 Support)
+
+***N/B:*** The repository's `.cirrus.yml` **must** define a
+**`EC2_INST_TYPE`** env. var. for any EC2-based tasks or globally.
+
+### Hack-script requirements
+
+1. Somewhere within the script the string `get_ci_vm APIv2` *must* be present.
```
Comment (edsantiago) [inline get_ci_vm/README.md:144]: If I'm reading `entrypoint.sh` correctly, the actual requirement is `# get_ci_vm APIv2` (hash / comment). If so, please adjust this text (and line 87 above).

**Review by edsantiago — COMMENTED** [review]
A few suggestions for clarity. I don't have the background to review it all, but this definitely helps, thank you!

File: get_ci_vm/README.md (hunk 1 above)
Comment (cevich) [reply get_ci_vm/README.md:26]: Good catch, I did a search/replace to add the italics and missed that case.  I'll fix it.

File: get_ci_vm/README.md (hunk 2 above)
Comment (cevich) [reply get_ci_vm/README.md:98]: I see what you mean, and am not sure how to fix it.  They're all used and valid/required in different contexts, zero, non-zero, `1` and `2`.  Hmmm, I'll have to do some thinking of how to word this, or maybe I should add a truth-table so it's more clear.

File: get_ci_vm/README.md (hunk 6 above)
Comment (cevich) [reply get_ci_vm/README.md:132]: good idea, thanks.

File: get_ci_vm/README.md (hunk 7 above)
Comment (cevich) [reply get_ci_vm/README.md:144]: correct, thanks I'll fix it.

File: get_ci_vm/README.md (hunk 2 above)
Comment (cevich) [reply get_ci_vm/README.md:98]: Okay, made some updates that hopefully will help make this more clear.

File: get_ci_vm/README.md (hunk 3 above)
Comment (cevich) [reply get_ci_vm/README.md:108]: This check is done in the hack script itself, [example from podman.](https://github.com/containers/podman/blob/main/hack/get_ci_vm.sh#L31-L53)  I'll see if I can make this more clear in the docs.

**Comment by edsantiago** [comment]
/lgtm

Thanks!

**Pull request merged**